tests/unit/useOpsInbox.test.ts

```
@@ -0,0 +1,381 @@
+// Build a chainable + thenable mock for list queries.
+// Supabase query builders are thenable (have .then()) so `await q` works,
+// but you can also keep calling .eq() / .order() etc. before awaiting.
+function buildListChain(resolvedValue: { data: unknown; error: unknown; count?: number | null }) {
+  const chain: Record<string, unknown> = {};
+  chain.select = vi.fn().mockReturnValue(chain);
+  chain.eq = vi.fn().mockReturnValue(chain);
+  chain.order = vi.fn().mockReturnValue(chain);
+  chain.limit = vi.fn().mockReturnValue(chain);
+  chain.update = vi.fn().mockReturnValue(chain);
+  // Make chain thenable so `await chain` resolves to resolvedValue
+  chain.then = vi.fn().mockImplementation((resolve: (v: unknown) => unknown) => {
```

Biome (2.3.14)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)